src/systemtask/SystemTask.cpp

```diff
@@ -424,6 +424,16 @@ void SystemTask::Work() {
         case Messages::BatteryPercentageUpdated:
           nimbleController.NotifyBatteryLevel(batteryController.PercentRemaining());
           break;
+        case Messages::LowBattery: {
+          Pinetime::Controllers::NotificationManager::Notification notif;
+          std::array<char, Pinetime::Controllers::NotificationManager::MessageSize + 1> message {
+            "Low Battery\0Charge your watch to prevent data loss.\0"};
+          notif.message = std::move(message);
+          notif.size = 53;
```

Comment:
```suggestion
          constexpr char message[] = "Low Battery\0Charge your watch to prevent data loss.\0";
          constexpr size_t messageSize = std::min(sizeof(message), Pinetime::Controllers::NotificationManager::MaximumMessageSize());
          std::memcpy(notif.message.data(), message, messageSize);
          notif.size = messageSize;
```

With this, you avoid hard-coding the size of the string (we'll most probably forget to update it if we change the message). It also checks that the content of the message fits into the notification buffer.

Reply:
Thank you! This is much better 😊